### ISSUE TITLE
Spellcheck not seeing '-'

### ISSUE THREAD

**Issue by moritz-t-w** [ONLYOFFICE/DesktopEditors#1532]
### This issue is unique.

- [X] I have used the [search tool](https://github.com/ONLYOFFICE/DesktopEditors/issues?q=) and did not find an issue describing my bug.

### Operating System

Linux (DEB package)

### Version information

8.0.1.31

### Expected Behavior

German spellcheck can see the whole word including a trailing `-` character

### Actual Behavior

![Kommunikations- > Kommunikations--](https://github.com/ONLYOFFICE/DesktopEditors/assets/90388353/be6ffefd-5e6d-41b1-aa17-94f73d35d176)
![Kommunikations-- > Kommunikations---](https://github.com/ONLYOFFICE/DesktopEditors/assets/90388353/76b964ae-6e84-433b-afdd-d6c3c5070e32)
![Kommunikations--- > Kommunikations----](https://github.com/ONLYOFFICE/DesktopEditors/assets/90388353/3b300ae0-3302-4bdd-8fd3-88c380b266c7)
It just keeps adding - and not seeing any existing -

### Reproduction Steps

1. Set spellcheck to "Deutsch (Schweiz)"
2. Write something like "Kommunikations-"
3. Correct it with spellcheck to what you already wrote
4. See it still underlined, now with two trailing "-"

**Comment by HectorB-2020** [comment]
@moritz-t-w, hello! I'm collecting issues related to multi-language documents, not affiliated with Only Office, mind you.
Since you reported an issue involving various languages, I thought about sharing them with you. In my view the most annoying is inability to limit list of laguages.
- https://github.com/ONLYOFFICE/DesktopEditors/issues/802
- https://github.com/ONLYOFFICE/DesktopEditors/issues/273
- https://github.com/ONLYOFFICE/DocumentServer/issues/2142
- https://github.com/ONLYOFFICE/DesktopEditors/issues/626
- https://github.com/ONLYOFFICE/DesktopEditors/issues/527
- https://github.com/ONLYOFFICE/DesktopEditors/issues/437
- https://github.com/ONLYOFFICE/DesktopEditors/issues/356
- https://github.com/ONLYOFFICE/DesktopEditors/issues/224
- https://github.com/ONLYOFFICE/DocumentServer/issues/576
- https://github.com/ONLYOFFICE/DesktopEditors/issues/1449
- https://github.com/ONLYOFFICE/DocumentServer/issues/2140
- https://github.com/ONLYOFFICE/DocumentServer/issues/2326
- https://github.com/ONLYOFFICE/DesktopEditors/issues/1223
- https://github.com/ONLYOFFICE/DesktopEditors/issues/1124

**Comment by moritz-t-w** [comment]
@HectorB-2020 This issue happens in single language docs too

**Comment by maruacat** [comment]
Hello @moritz-t-w 
Please tell me, in this situation, is the “-” symbol at the end of a word part of the word? Could you explain to me a little more what the correct spell check behavior would be in this situation?

**Comment by moritz-t-w** [comment]
So in german you can shorten something like this
> The foreground and the background

To this
> The fore- and background

So the correction it's suggesting is correct, the problem is just that it doesn't take the - into account when generating corrections.

**Comment by maruacat** [comment]
> So in german you can shorten something like this
> 
> > The foreground and the background
> 
> To this
> 
> > The fore- and background
> 
> So the correction it's suggesting is correct, the problem is just that it doesn't take the - into account when generating corrections.

So after selecting the word "Kommunikations-" correction, the result should be that the word is no longer underlined in red and there should be only one dash left after the word, right?

**Comment by moritz-t-w** [comment]
> So after selecting the word "Kommunikations-" correction, the result should be that the word is no longer underlined in red and there should be only one dash left after the word, right?

Yes and if I wrote "Komunikations-" it should correct that to "Kommunikations-" aswell, again taking the - into account

**Comment by maruacat** [comment]
> > So after selecting the word "Kommunikations-" correction, the result should be that the word is no longer underlined in red and there should be only one dash left after the word, right?
> 
> Yes and if I wrote "Komunikations-" (with a typo) it should correct that to "Kommunikations-" aswell, again taking the "-" into account

Sorry, I have another one question. Did I understand correctly that the word "Kommunikations-" that you wrote in Reproduction Steps is written correctly and therefore it should not have been emphasized as incorrect at all? the same goes for other grammatically correct words with a hyphen at the end

**Comment by moritz-t-w** [comment]
> Did I understand correctly that the word "Kommunikations-" that you wrote in Reproduction Steps is written correctly and therefore it should not have been emphasized as incorrect at all? the same goes for other grammatically correct words with a hyphen at the end

Yes.

**Comment by maruacat** [comment]
Hello @moritz-t-w 
Thanks for your help, everything is clear to me now. I can confirm this is a bug, so I've created issue #67144.
Have a good day!

**Comment by Rita-Bubnova** [comment]
The issue is actual in DocumentServer.
